lib/plugins/inflightRequestThrottle.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+var assert = require('assert-plus');
+var ServiceUnavailableError = require('restify-errors').ServiceUnavailableError;
+var defaultResponse = new ServiceUnavailableError('resource exhausted');
+
+/**
+ * inflightRequestThrottle
```

Comment:
You may want to specify both here and in docs that users should register this plugin *first* as order matters and you want the throttle check to occur as early as possible.